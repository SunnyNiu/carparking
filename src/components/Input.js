import React, { useRef } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { DEFAULT_COMMANDS } from '../functions/const';
import { startCommandSequence } from '../sagas/actions';

const style = {
  maxHeight: '110px',
  minHeight: '80px',
};

const Input = ({ isRunning, dispatch }) => {
  const myRef = useRef(null);

  const changeCarLocation = () => {
    const array = myRef.current.value.trim().split('\n');
    dispatch(startCommandSequence(array));
  };
  return (
    <div>
      <label>
        Please input commands:
        <textarea ref={myRef} defaultValue={DEFAULT_COMMANDS} style={style} />
      </label>
      <button
        type="submit"
        value="Submit"
        onClick={changeCarLocation}
        disabled={isRunning}
      >
        Submit
      </button>
    </div>
  );
};

Input.propTypes = {
  isRunning: PropTypes.bool.isRequired,
  dispatch: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  isRunning: state.app.isRunning,
});
export default connect(mapStateToProps)(Input);
