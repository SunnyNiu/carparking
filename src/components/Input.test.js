import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Input from './Input';

const mockStore = configureStore([]);

describe('Input', () => {
  const store = mockStore({
    app: {
      location: {
        x: 0,
        y: 4,
        facing: 'WEST',
      },
      output: '',
    },
  });

  const wrapper = mount(
    <Provider store={store}>
      <Input />
    </Provider>
  );

  it('Input snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
