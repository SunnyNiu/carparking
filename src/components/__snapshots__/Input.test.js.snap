// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> component Input snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Input)>
    <Input
      dispatch={[Function]}
    >
      <div>
        <label>
          Please input commands:
          <textarea
            defaultValue="PLACE 1,2,EAST
MOVE
MOVE
LEFT
MOVE
REPORT"
            style={
              Object {
                "maxHeight": "110px",
                "minHeight": "80px",
              }
            }
          />
        </label>
        <button
          onClick={[Function]}
          type="submit"
          value="Submit"
        >
          Submit
        </button>
      </div>
    </Input>
  </Connect(Input)>
</Provider>
`;
